### PR TITLE
Removes PIXI global object in ES and CJS bundles to enable tree-shaking

### PR DESCRIPTION
**Pull request by bigtimebuddy**
### Overview

Partially addresses #5397.

This change removes `PIXI` from window object when defined in **es** and **cjs**  bundle builds. It's now _only_ enabled for the browser build. This will help with tree-shaking using bundlers like Webpack and Rollup since everything doesn't hang off a single global object.

#### Changes

* Changes browser build from **umd** format to **iife** format
* Removes `PIXI` global being defined in **es** or **cjs** formats for bundles
* Fixes tests that were still relying on `PIXI`
* Implements an new API to use deprecations for **es** and **cjs** builds

#### Pre-Merge Checklist

- [x] Tests and/or benchmarks are included
- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: bundles/pixi.js-legacy/test/index.js [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
-
 describe('PIXI', function ()
 {
+    before(function (done)
+    {
+        this.head = document.querySelector('head');
+        this.script = document.createElement('script');
+        this.script.onload = () => done();
+        this.script.src = require.resolve('../dist/pixi-legacy');
+        this.head.appendChild(this.script);
```
Comment (englercj) [inline bundles/pixi.js-legacy/test/index.js]: What is the benefit of putting these on `this` rather than just having them be vars in the describe scope?

Comment (bigtimebuddy) [reply bundles/pixi.js-legacy/test/index.js]: No benefit, mostly best practice. In general I like to be able to setup and tear down in mochas conventions

**Comment by bigtimebuddy** [comment]
The PIXI global is set via Rollups “output.name” field.

File: bundles/pixi.js-legacy/test/index.js (hunk 1 above)
Comment (englercj) [reply bundles/pixi.js-legacy/test/index.js]: Sorry I didn't meant to not set them here, just that the vars would be scoped to the describe:

```js
describe(function ()
{
    let head, script;

    before(function () { head = ... });
    after(function () { ... });
});
```

Doesn't matter, just curious.

**Comment by englercj** [comment]
Thanks @bigtimebuddy, I think this is good to go.

**Pull request merged**